modules/workers/src/main/java/org/apache/ignite/internal/worker/ThreadAssertions.java

```diff
@@ -64,20 +59,22 @@ public static void assertThreadAllowsToRead() {
     public static void assertThreadAllowsTo(ThreadOperation requestedOperation) {
         Thread currentThread = Thread.currentThread();
 
-        if (BLACKLISTED_THREAD_NAMES.matcher(currentThread.getName()).matches()) {
-            return;
-        }
-
         if (!(currentThread instanceof ThreadAttributes)) {
+            if (PublicApiThreading.executingSyncPublicApi()) {
+                // Allow everything if we ride a user thread while executing a public API call.
+
+                return;
+            }
+
             LOG.warn("Thread {} does not have allowed operations", trackerException(), currentThread);
```

Comment:
Why do we need both a warning and an exception?

Reply:
Some errors/exceptions get swallowed, so we can't rely on the `AssertionError` to track the problem that surfaced. Having it logged explicitly gives this guarantee.

I removed the second (tracking) exception though.